tests/src/JIT/Performance/CodeQuality/BenchI/QuickSort/QuickSort.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Xunit.Performance;
+using System;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+[assembly: OptimizeForBenchmarks]
+[assembly: MeasureInstructionsRetired]
+
+public static class QuickSort
+{
+
+#if DEBUG
+    public const int Iterations = 1;
+#else
+    public const int Iterations = 80000;
+#endif
+
+    const int MAXNUM = 200;
+    const int MODULUS = 0x20000;
+    const int C = 13849;
+    const int A = 25173;
+    static int s_seed = 7;
+
+    static int Random(int size) {
+        unchecked {
+            s_seed = s_seed * A + C;
+        }
+        return (s_seed % size);
+    }
+
+    static void Quick(int lo, int hi, int[] arr) {
+
+        int i, j;
+        int pivot, temp;
+
+        if (lo < hi) {
+            for (i = lo, j = hi, pivot = arr[hi]; i < j;) {
+                while (i < j && arr[i] <= pivot){
+                    ++i;
+                }
+                while (j > i && arr[j] >= pivot) {
+                    --j;
+                }
+                if (i < j) {
+                    temp = arr[i];
+                    arr[i] = arr[j];
+                    arr[j] = temp;
+                }
+            }
+            
+            // need to swap the pivot and a[i](or a[j] as i==j) so
+            // that the pivot will be at its final place in the sorted array
+
+            if (i != hi) {
+                temp = arr[i];
+                arr[i] = pivot;
+                arr[hi] = temp;
+            }
+            Quick(lo, i - 1, arr);
+            Quick(i + 1, hi, arr);
+        }
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static bool Bench() {
+        int[] buffer = new int[MAXNUM];
+        for (int j = 0; j < MAXNUM; ++j) {
+            int temp = Random(MODULUS);
+            if (temp < 0){
+                temp = (-temp);
+            }
+            buffer[j] = temp;
+        }
+        Quick(0, MAXNUM - 1, buffer);
+        return true;
```

Comment:
Would be straightforward to add checking code that the array was sorted like the other sort benchmarks.